src/api/schemas.py

```diff
@@ -137,7 +137,7 @@ class ExternalSiteUserRequest(RequestBase):
     first_name: Optional[str] = Field(None, max_length=64)
     last_name: Optional[str] = Field(None, max_length=64)
     email: str = Field(..., max_length=48)
-    specializations: Optional[str] = Field(...)
+    specializations: Optional[list[int]] | None
```

Comment:
Конструкция `Optional[...]` и `... | None` обозначают одно и тоже. Ты их как то объединил...
```suggestion
    specializations: list[int] | None
```

Также, как я описал в задаче, мы будем принимать от клиента строку, примерно как: `"1, 2, 3"` . Нужно добавить валидатор, чтобы с ней также работала.

Reply:
Добавил валидатор строки через регулярку